### PR TITLE
Add new OCR tests with RPC node failures

### PR DESCRIPTION
**Pull request by lukaszcl**
- `TestOCRSoak_RPCDownForHalfCLNodes` simulates a network chaos by bringing down network to RPC node for 50 percent of Chainlink Nodes
- `TestOCRSoak_RPCDownForAllCLNodes` simulates a network chaos by bringing down network to RPC node for all Chainlink Nodes
 
Example run: https://grafana.ops.prod.cldev.sh/goto/4en1jl8IR?orgId=1)

### CODE REVIEW

File: integration-tests/soak/ocr_test.go [hunk 1]
```diff
@@ -22,58 +30,183 @@ func TestOCRSoak(t *testing.T) {
...
+// TestOCRSoak_RPCDownForAllCLNodes simulates a network chaos by bringing down network to RPC node for all Chainlink Nodes
+func TestOCRSoak_RPCDownForAllCLNodes(t *testing.T) {
+	config, err := tc.GetConfig(t.Name(), tc.OCR)
+	require.NoError(t, err, "Error getting config")
+	require.True(t, config.Network.IsSimulatedGethSelected(), "This test requires simulated geth")
+
+	namespace := fmt.Sprintf("%s-%s", "soak-ocr-simulated-geth", uuid.NewString()[0:5])
+	chaos, err := gethNetworkDownChaos(GethNetworkDownChaosOpts{
+		DelayCreate: time.Minute * 2,
+		Duration:    time.Minute * 5,
```
Comment (Tofel) [inline integration-tests/soak/ocr_test.go:81]: we are sure we don't want these values to be configurable via TOML?

File: integration-tests/soak/ocr_test.go [hunk 2]
```diff
@@ -22,58 +30,183 @@ func TestOCRSoak(t *testing.T) {
...
+			Selector: v1alpha1.PodSelectorSpec{
+				GenericSelectorSpec: v1alpha1.GenericSelectorSpec{
+					Namespaces:     []string{namespace},
+					LabelSelectors: map[string]string{"app": "chainlink-0"},
+				},
+			},
+			Mode:  v1alpha1.FixedPercentMode,
+			Value: "50",
+		},
+	})
```
Comment (Tofel) [inline integration-tests/soak/ocr_test.go:127]: this and previous chaos options differ only in 2 values, why not extract common part?

File: integration-tests/testconfig/ocr/ocr.toml [hunk 3]
```diff
@@ -171,4 +171,26 @@ selected_networks=["Anvil"]
...
+test_duration="15m"
+[TestOCRSoak_RPCDownForAllCLNodes.OCR.Soak]
+ocr_version="1"
+number_of_contracts=2
+time_between_rounds="1m"
+[TestOCRSoak_RPCDownForAllCLNodes.Network]
+selected_networks=["simulated"]
+
+[TestOCRSoak_RPCDownForHalfCLNodes.Common]
+chainlink_node_funding = 100
```
Comment (Tofel) [inline integration-tests/testconfig/ocr/ocr.toml]: this is excessive, 0.5 is enough on Sepolia for a 24h test :-)

File: integration-tests/soak/ocr_test.go (hunk 1 above)
Comment (lukaszcl) [reply integration-tests/soak/ocr_test.go:81]: @Tofel 

I had the same idea at the beginning (see my previous commits). But, now I see more downsides like:

1. We would need to create custom fields or structs for this in TOML
2. We would probably need to put other params as well like chaos description, chaos selectors, chaos mode, etc. (see chaos definition in tests). Do you think it it is useful for anyone to expose it in TOML?

File: integration-tests/soak/ocr_test.go (hunk 2 above)
Comment (lukaszcl) [reply integration-tests/soak/ocr_test.go:127]: @Tofel it's not only these values. The common part is extracted in gethNetworkDownChaos(). Keeping `GethNetworkDownChaosOpts` in the tests makes them more easy to understand. E.g. how long chaos will take, etc.

File: integration-tests/testconfig/ocr/ocr.toml (hunk 3 above)
Comment (lukaszcl) [reply integration-tests/testconfig/ocr/ocr.toml]: @Tofel I pushed a new commit with the fix [Update chainlink_node_funding](https://github.com/smartcontractkit/chainlink/pull/13624/commits/45709f52f4d2b738be7f263ad0602723b74ad07a)

File: integration-tests/soak/ocr_test.go (hunk 1 above)
Comment (Tofel) [reply integration-tests/soak/ocr_test.go:81]: Hard to say at this point how people will use this, so maybe let's leave as-is until someone requests it?

File: integration-tests/testsetups/ocr.go [hunk 4]
```diff
@@ -88,10 +91,25 @@ type OCRSoakTest struct {
...
+
+func WithNamespace(ns string) OCRSoakTestOption {
+	return func(c *OCRSoakTest) {
+		c.namespace = ns
+	}
 }
 
 // NewOCRSoakTest creates a new OCR soak test to setup and run
-func NewOCRSoakTest(t *testing.T, config *tc.TestConfig, forwarderFlow bool) (*OCRSoakTest, error) {
+func NewOCRSoakTest(t *testing.T, config *tc.TestConfig, forwarderFlow bool, opts ...OCRSoakTestOption) (*OCRSoakTest, error) {
```
Comment (Tofel) [inline integration-tests/testsetups/ocr.go]: small extra: what about making `forwarderFlow` also an opt? would be easier to understand without checking this method's signature to check what that bool is about.

**Review by Tofel — DISMISSED** [review]

File: integration-tests/soak/ocr_test.go (hunk 1 above)
Comment (lukaszcl) [reply integration-tests/soak/ocr_test.go:81]: @Tofel yeah, I think this approach makes sense. It's very easy to update if needed. It's a "two-way door" decision, changeable and reversible according to Bezos's framework that I really like https://coda.io/@hales/jeff-bezos-make-fast-decisions

File: integration-tests/testsetups/ocr.go (hunk 4 above)
Comment (lukaszcl) [reply integration-tests/testsetups/ocr.go]: @Tofel yes, this makes sense. Let me do it!

Comment (lukaszcl) [reply integration-tests/testsetups/ocr.go]: @Tofel fixed in [Make forwarderFlow an opt](https://github.com/smartcontractkit/chainlink/pull/13624/commits/77c9fd7cebcb64c7e470acf14f0be65aaf0f7475)

**Review by Tofel — APPROVED** [review]

**Review by davidcauchi — APPROVED** [review]

**Review by anirudhwarrier — APPROVED** [review]

**Pull request merged**